### PR TITLE
Demo app with toolbar, filter, and sort components

### PR DESCRIPTION
**Pull request by dlabrecq**
Added toolbar, filter, filter-field, filter-results, and Dana's sort component from Angular Patternfly. These separate components were built together in order to be consumed by the toolbar. This Angular2 conversion includes all existing Patternfly unit tests as well as examples.

I also created a demo app, built using webpack, to showcase new components. To view the demo, run "npm run start" and point your browser to http://localhost:3001.

Note that the styles are a little off for the toolbar view icons. The build also occasionally generates a "stack level too deep" error with sass-loader -- a rebuild often clears that up. Will need to evaluate how the CSS is built with Patternfly.

There is more work to be done here, but this is a good stopping point as the number of changes are getting large.

Toolbar Example:
screen shot 2017-02-24 at 1 00 54 am

Filter Example:
screen shot 2017-02-24 at 1 01 56 am

Sort Example:
screen shot 2017-02-24 at 1 01 31 am

### CODE REVIEW

**Comment by dlabrecq** [comment]
FYI @dgutride

File: package.json [hunk 1]
```diff
@@ -157,7 +162,7 @@
     "rollup-plugin-angular-aot": "^0.0.4",
     "run-sequence": "^1.2.2",
     "rxjs": "5.0.0-beta.12",
-    "sass-loader": "4.0.2",
+    "sass-loader": "6.0.2",
```
Comment (joshuawilson) [inline package.json]: I've run into several problems with sass-loader 6.x, I have had to down grade to 4.1.1. Is there a reason to use 6.x?

File: package.json [hunk 2]
```diff
@@ -79,6 +81,7 @@
     "@angular/http": "2.2.3",
     "@angular/platform-browser": "2.2.3",
     "@angular/platform-browser-dynamic": "2.2.3",
+    "@angular/router": "3.1.1",
```
Comment (joshuawilson) [inline package.json]: router can match the rest of angular. you can use 3.2.3

File: src/app/app.module.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    SortExampleModule,
+    ToolbarExampleModule
+  ],
+  declarations: [
+    AppComponent
+  ],
+  providers: [],
+  bootstrap: [ AppComponent ]
+})
+export class AppModule { }
```
Comment (joshuawilson) [inline src/app/app.module.ts]: Why did you add an app.module? For the demo of the examples?

File: src/app/treelist/treelist.component.spec.ts [hunk 4]
```diff
@@ -13,16 +13,14 @@ import { By }           from '@angular/platform-browser';
...
-import { WorkItem } from '../../models/work-item';
 
 describe('Treelist component - ', () => {
   let comp: TreeListComponent;
   let fixture: ComponentFixture<TreeListComponent>;
   let el: DebugElement;
-  let fakeUserList: User[];
-  let fakeWorkItem: WorkItem;
-  let fakeWorkItems: WorkItem[] = [];
+  let fakeUserList: any[];
```
Comment (joshuawilson) [inline src/app/treelist/treelist.component.spec.ts]: nice catch, don't want workitems in there :)

File: widgets.module.ts [hunk 5]
```diff
@@ -41,8 +44,11 @@ import { AlmTrim } from './src/app/pipes/alm-trim';
     ContainerTogglerModule,
     DialogModule,
     DropdownModule,
+    FiltersModule,
```
Comment (joshuawilson) [inline widgets.module.ts]: if you want these exported don't forget to add them to the 'index.ts' file

File: package.json (hunk 1 above)
Comment (dlabrecq) [inline package.json]: I was hoping to get rid of the "stack level too deep" issue, but we can go back to 4.1.1.

File: package.json (hunk 2 above)
Comment (dlabrecq) [inline package.json]: Yes

File: src/app/app.module.ts (hunk 3 above)
Comment (dlabrecq) [inline src/app/app.module.ts]: Only for the demo. This isn't included in the library build, only consumed in webpack when the examples are run.

File: widgets.module.ts (hunk 5 above)
Comment (dlabrecq) [inline widgets.module.ts]: Right. I did that for toolbar, but forgot filter and sort.

**Comment by dlabrecq** [comment]
Added exports to index.ts and modified package versions per code review comments.

**Comment by joshuawilson** [comment]
@dlabrecq there are failing tests.

**Comment by joshuawilson** [comment]
@dlabrecq when I ran the demo with "npm start" I got the following error.

ERROR in ./~/css-loader!./~/sass-loader!./src/app/filters/filter.component.scss
Module build failed: 
  @include navbar-vertical-align($line-height-computed);
                  ^
      Stack level too deep
      in /home/josh/git/fabric8-ui/ngx-widgets/src/assets/vendor/patternfly/stylesheets/bootstrap/_navbar.scss (line 360, column 12)
 @ ./src/app/filters/filter.component.scss 4:13-138
 @ ./src/app/filters/filter.component.ts
 @ ./src/app/filters/filters.module.ts
 @ ./src/app/filters/examples/filter-example.module.ts
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi (webpack)-dev-server/client?http://0.0.0.0:3001 ./src/main.ts
